microcosm/opaque.py

```diff
@@ -29,6 +29,39 @@
 )
 
 
+class LowercaseDict(dict):
```

Comment:
Easier to inherit from MutableMapping or UserDict. this way you only have to override the __setitem__ and not the other mutation methods

Reply:
dict is using C implementation under the hood, so breaks some conventions

Reply:
I'm sticking to the `dict`, because it's supposed to be faster and works with `isinstance`.